### PR TITLE
Dash board calendar picker

### PR DESCRIPTION
**Pull request by djnunez-aot**
-added calendar picker to filter from & to date for data

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
         "@mui/lab": "^5.0.0-alpha.91",
         "@mui/material": "^5.6.1",
         "@mui/x-date-pickers": "^5.0.5",
+        "@mui/x-date-pickers-pro": "^6.4.0",
```
Comment (saravanpa-aot) [inline met-web/package.json:17]: what does the pro do?
what licence it has?

Comment (djnunez-aot) [reply met-web/package.json:17]: Pro is another material ui sub library, there's no licensing issue here though the issue is only when we use the DateRangeCalendar component.

Comment (djnunez-aot) [reply met-web/package.json:17]: the pro component is just cleaner and does not require the render input prop

Comment (saravanpa-aot) [reply met-web/package.json:17]: can you point to some links on the licensing? also pro vs pickers in terms of the 'pro' term?

Comment (djnunez-aot) [reply met-web/package.json:17]: https://mui.com/pricing/ this is the licensing chart 

the pro library picker:
`   <DatePicker
                                    value={fromDate}
                                        onChange={(newDate: Dayjs | null) => setFromDate(newDate)}
                                    />`
 The non pro picker:
 `     <DatePicker
                                    inputFormat="MM/DD/YYYY"
                              value={fromDate}
                                        onChange={(newDate: Dayjs | null) => setFromDate(newDate)}
                                    renderInput={(params) => <TextField {...params} />}
                                />`

Comment (saravanpa-aot) [reply met-web/package.json:17]: so are we using a non-free version? 

@jadmsaadaot thoughts here?

Comment (djnunez-aot) [reply met-web/package.json:17]: no this is still the free version. It's only certain components in the library that are not free.

Comment (djnunez-aot) [reply met-web/package.json:17]: Yea that sounds good ill revert to the regular date picker then and remove the pro library

**Review by tiagograf-AOT — APPROVED** [review]
It looks like same changes as the other PR?!

**Comment by djnunez-aot** [comment]
The other pr is suppose to be a 1 file change. I was going to rebase once the other one got merged since they are separate tasks. I could just do put them all in one though

File: met-web/src/components/publicDashboard/SubmissionTrend/SubmissionTrend.tsx [hunk 2]
```diff
@@ -46,14 +67,30 @@ const SubmissionTrend = ({ engagement, engagementIsLoading }: SubmissionTrendPro
     };
 
     useEffect(() => {
-        fetchData();
+        fetchData().catch((error) => {
+            console.error(error);
+        });
     }, [engagement.id, chartBy]);
 
+    useEffect(() => {
```
Comment (jadmsaadaot) [inline met-web/src/components/publicDashboard/SubmissionTrend/SubmissionTrend.tsx:75]: do we have to use two useEffects to call the same function, can we do better?

Comment (jadmsaadaot) [reply met-web/src/components/publicDashboard/SubmissionTrend/SubmissionTrend.tsx:75]: I think if we just add the dependencies [fromDate, toDate] to the first one it should work, can you please try that?

**Review by jadmsaadaot — CHANGES_REQUESTED** [review]
Please try adding the  [fromDate, toDate] to the first useEffect and see if that works instead of using two use Effects, you wouldn't need the if (fromDate && toDate) in this case I think too

**Review by jadmsaadaot — CHANGES_REQUESTED** [review]
Please try adding the  [fromDate, toDate] to the first useEffect and see if that works instead of using two use Effects, you wouldn't need the if (fromDate && toDate) in this case I think too

**Review by jadmsaadaot — CHANGES_REQUESTED** [review]
Please try adding the  [fromDate, toDate] to the first useEffect and see if that works instead of using two use Effects, you wouldn't need the if (fromDate && toDate) in this case I think too

File: met-web/src/components/publicDashboard/SubmissionTrend/SubmissionTrend.tsx (hunk 2 above)
Comment (djnunez-aot) [reply met-web/src/components/publicDashboard/SubmissionTrend/SubmissionTrend.tsx]: done please review again

**Review by jadmsaadaot — APPROVED** [review]

**Pull request merged**